pkg/syncer/metadatasyncer.go

```diff
@@ -2637,6 +2638,11 @@ func podAdded(obj interface{}, metadataSyncer *metadataSyncInformer) {
 			log.Warnf("podAdded: unrecognized new object %+v", obj)
 			return
 		}
+		if len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+			log.Info("podAdded: In-tree vSphere volume are not supported in a multi VC setup. " +
```

Comment:
this is not correct for regular csi volumes.
we need to error out when pod has in-tree volume and multi vcenter deployment is detected and pod metdata is being pushed.

Reply:
are we pushing pod metadata for regular csi volume using `podUpdated` method?
if yes then is `podAdded` is exclusively used for pushing metadata for inline vSphere volumes. in that case i think, proposed change is fine. Can you confirm?